iorgen/check_input.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright 2018 Sacha Delanoue
+# Copyright 2019 Matthieu Moatti
+"""Check that a YAML input is well formed"""
+
+import re
+from typing import Any, Dict, List, TextIO, Union
+
+import yaml
+
+import warnings
+
+from iorgen.types import Input, Struct, Type, Variable
+
+class Counter:
+    errors = 0
+    warnings = 0
+
+def increase_error() -> None:
+    """Increase error counter"""
+    Counter.errors += 1
```

Comment:
This is still a global variable. What about making `increase_error` part of a class, so that it could do `self.errors += 1`?

Reply:
Well I can do an object that travels inside the functions.

Reply:
That works too. Though if every function has the same object in argument, then maybe a class is more appropriated.

Reply:
I must admit that I don't see what you mean there.

Reply:
```python
class Checker:
    def __init__(self):
        self.errors = 0

    def increase_error(self):
       self.errors += 1

    […]

    def input_from_dict(dic):
        if not_good(dic):
            self.increase_error()
        return good_input

def parse_input(stream):
    return Checker().input_from_dict(yaml.load(stream))
```